darwin/dataset/remote_dataset_v2.py

```diff
@@ -47,6 +48,19 @@
 if TYPE_CHECKING:
     from darwin.client import Client
 
+from tenacity import (
+    RetryCallState,
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential_jitter,
+)
+
+
+def before_sleep(retry_state: RetryCallState):
```

Comment:
would call this something more exact to what it does: `log_rate_limit_exceeded` or similar